source/common/network/connection_impl.cc

```diff
@@ -128,6 +141,7 @@ void ConnectionImpl::close(ConnectionCloseType type) {
     // Create and activate a timer which will immediately close the connection if triggered.
     // A config value of 0 disables the timeout.
     if (delayed_close_timeout_set) {
+      ASSERT(connection_stats_->delayed_close_timeouts_ != nullptr);
```

Comment:
Sorry, I meant in setDelayedCloseTimeout.I don't want someone to enable this feature on a connection and then only later get an assertion about this stat.

Reply:
`connection_stats_` is set via `setConnectionStats()`, so asserting on the counter != nullptr in `setDelayedCloseTimeout()` will introduce an ordering dependency (which is not currently met). I think it would be preferable to keep these two functions independent of each other.

Unfortunately, initialization of `ConnectionImpl`s is spread out between codec creation (allocation and call to `setDelayedCloseTimeout`) and `ConnectionManagerImpl::initializeReadFilterCallbacks()` where `setConnectionStats()` is called. Therefore, adding an `initFinished()` function where the ASSERT could be safely issued will require some work.

I can mark this with a TODO so we can move it into the appropriate place when such a function is added. What do you think?

Reply:
Or just make setting connection_state_->delayed_close_timeouts_ optional? (E.g. don't assert, just don't increment if it's null?)

Reply:
Ok, done.